contracts/validator/ValidatorSet.sol

```diff
@@ -50,4 +50,74 @@ contract ValidatorSet {
      *   - less than MAX_FUTURE_END_HEIGHT: for logged out validators.
      */
     mapping(address => uint256) public validatorEndHeight;
+
+
+    /* Internal functions  */
+
+    /**
+     * @notice It is for inserting validators into the validator set.
+     *
+     *      Function requires :
+     *          - Validator address must not be 0.
+     *          - Validator must not be not part of this core.
+     *          - Validator must have non-zero begin height.
+     *
+     * @param _validator Validator address.
+     * @param _beginHeight Begin height for the validator.
+     */
+    function insertValidator(address _validator, uint256 _beginHeight) internal {
+        require(
+            _validator != address(0),
+            "Validator must not be null address."
+        );
+        require(
+            validatorEndHeight[_validator] == 0,
+            "Validator must not already be part of this core."
+        );
+        require(
+            validatorBeginHeight[_validator] == 0,
+            "Validator must not have a non-zero begin height"
+        );
```

Comment:
Suggestion: above both the `requires` are used for the same purpose, maybe it can be combined.

Reply:
Thank you. As both had same purposer, removed **require** for **validatorEndHeight**.

Reply:
the combination of these two requirements is an `OR` condition. see https://github.com/benjaminbollen/mosaic-1/blob/protocore/contracts/validator-set/ValidatorSet.sol#L100